### ISSUE TITLE
Uganda2011

### ISSUE THREAD

**Issue by robin-wang** [wengxyu1030/AIS#10]
WORKFLOW DOC

{FOR REVIEWERS}
- [x] They can run the code smoothly on the reviewer's end, where the microdata could be generated without error
- [ ] Review and validate the quality checking output

{FOR CODERS}
A. General Procedural Steps

- [x] Rebase current branch based on Master, ensure your branch is the latest.
Tutorial: Rebasing your project branch onto another branch
https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/syncing-your-branch#rebasing-your-project-branch-onto-another-branch

- [x] Run script without error
- [x] Pass internal quality checking (the embedded quality check result: successfully produced value for the indicators and check if there're flags that the indicator value produced is too off from the published data) 
- [x] Document as comments any notable changes/caveats (if any)
- [x] When Fully Completed Survey Edits, proceed to record hours worked: https://forms.gle/tLh7fxCMMsLobGHA6

B. Checklist and Guidance

- [x] ANC: 【birth.dta not available. Related variables are all coded as NA in ind.dta】
  - Skill providers
    -  Checked in the regular express section in Programs 
    - **【m2a - m2m "prenatal care provider" are all NA, replaced by s302a-e, s302x-y "antenatal care provider" in the ind.dta. However, as the birth.dta is not able to be generated in the first place, the related variables in the final dataset still has 0 obs.】**
  - Tetanus Injection
    -  Number of Injection Before/During Pregnancy
    -  Recall period for last injection
    -  Relevant data comes other sections
  - Blood Pressure / Blood Sample / Urine Sample
    -  Samples (m42c/d/e) are sub-questions for "receive antenatal care" in questionnaire or not

- [x] Delivery Care: 【birth.dta not available. Related variables are all coded as NA in ind.dta】
  - c_facdel /  delivery institutions  / c_hospdel / delivery institutions
    -  Consistent with the regular expression section in Programs 
  - c_skin2skin
    -  Check if available; if, then note which variable
  - m15 skip condition
    -  Usually delivery location is a filter question for c_caesarean(m17) / stay(m61) / mother or child PNC; is it m15, if not which one, and list the conditions used for m15  if pattern is peculiar (normally (m15 = 11, 12 etc))

- [x] PNC: 【birth.dta not available. Related variables are all coded as NA in ind.dta】
  - child pnc
    -  See if child pnc is only avaliable when (m15=11,12,96) or (m15!=11,12,96)
    -  Check if Child PNC is designed to be from full sample, i.e not a subquestion within fertility section. If a subquestion, don't generate c_pnc_any and c_pnc_eff
   
- [x] Child Vaccination (search receive/month in birth.dta. if unknown vaccine pops up, search definition in DHS report)  : 【birth.dta not available. Related variables are all coded as NA in ind.dta】
  - BCG
  - DTP/DPT, DTaP, DTwP or pentavalent 1-3 against diphtheria, tetanus and pertussis
  - Polio 1-3 (IPV, OPV)
  - Measles/MMR
  - Other vaccines, if any,

- [x] Child Illness: 【birth.dta not available. Related variables are all coded as NA in ind.dta】
  - 20210510 For Providers, focus on the skilled providers rather than all
  - Diarrhea treatment provider
  - Check provider list being consistent with existing list (if not make changes in program)
  - Fever/Ari1/Ari2 treatment provider
  - Provider list being consistent with existing list
  - Fever/Fever Treatment/Cough/Cough Treatment from Full Sample
  - c_diarrhea_med, medicine and formal meds to treat are survey-specific. 20210510 focus on special medicines
  - c_diarrhea_medfor, medicine and formal meds to treat are survey-specific. 20210510 focus on special medicines
  - Validate c_ari can be generated or not

- [x] Child Mortality
  - General Observations

- [x] Sleeping Under Bednet
  - Adhere to program's logic, but please list any specific bednet related observations (tricky cases for presumed treated or the like)

- [x] Child Anthropometrics
  - General Observations (No need to generate "Wasting")

- [x] Reproductive Health
  - Contraceptive, check modern methods: 【v313 exist, but has 0 observation】
  - In Need - most updated version of logics in program (temporarily no actions needed)

- [x] Female Anthropometrics and Cancer Screening: 【not available】
  - Mammogram (X-ray, mamm...)
  - Pap Smear (Cervical vaginal cytology)
  - Note recall indicators for Mamm or Pap Smear (for example, s1012n used as filter for s1012m)

- [x] Hospitalisation, NCD & HIV
  - Hospitalisation, general observations: 【not available in any datasets】			
  - Hypertension treatment , blood pressure measure in the past or during the survey, for example check hm.dta/ind.dta/men.dta (generate a_bp_treat a_bp_sys a_bp_dial a_bp_meas a_hi_bp140 a_hi_bp140_or_on_med from ind.dta & men.dta by combining them): 【not available in any datasets】

    - Note from WB team for reference: "don't worry if you cannot find men.dta, not all surveys have it. if it is collected from ind.dta and men.dta seperately (some survey have female & male questionnarie, collecting these data seperately), you will need to combine these 2 dataset, generating these indicators and merge them back to hm.dta. If only from ind.dta or men.dta, i.e. only for female or male, don't generate these variables. If a new/survey-specific datafile occurs, it is highly likely that it contains info. for 13.do.  "
    
  - Diabetes treatment/measure in the past (for example, a_diab_treat): 【not available in any datasets】
  - HIV, general observations
    - HIV questionnaire might exsit in Section 4 of "HEALTH ISSUES QUESTIONNAIRE". If they don't ask hiv test results, they are not helpful. but please document these observations.
    - 【hiv03 test results 0 = negative, 1 = positive】

- [x] Cross-validate statistics in survey report, if applicable.
   - 【2011 AIDS indicator info sheet "HF41.pdf" shows that the total HIV prevalence is 7.3% among women and men aged 15-49. In the cleaned dataset, a_hiv_cat mean = 0.0708522, which is very close to the report. 】
   - 【All the other information is not covered in the survey report.】

- [x] Reference the NAME-SURVEY branch when your edits are finalised.
   - Refer to the github working branch created for [Name-SurveyName]

- [x] Update micro data in Onedrive

- [x] Examine Quality Checking Output Files
  - [x] Successfully produced the indicators:
        - 【Indicators are not generated, as there is no "AIS Uganda 2011" surveyid matched in the DHS or HEFPI datasets.】
        
<img width="659" alt="Screen Shot 2021-07-02 at 11 01 49 AM" src="https://user-images.githubusercontent.com/60246203/124312915-fa534980-db24-11eb-86be-b29335684034.png">
<img width="653" alt="Screen Shot 2021-07-02 at 11 02 09 AM" src="https://user-images.githubusercontent.com/60246203/124312923-fe7f6700-db24-11eb-841c-4b0c021f4f3d.png">

  - [x] Check if there's a flag that the indicator value produced is too off from the published data
      - 【Not generated.】

**Comment by yining-sun** [comment]
@robin-wang @wengxyu1030 

Dear both, 

Regarding the Uganda2011 survey, 
- There are no quality control indicators generated as the AIS survey id is not available in the quality check datasets. 
- About the antenatal care providers, the m2a-m variables do not have observations. Instead, I found s302a-y might be used to replace that. I didn't make changes in the current pull request #25 . Let me know if you would like to use s302a-y instead. 

Best, 
Yining

**Comment by wengxyu1030** [comment]
Dear Yining, 

Thanks for the pull request. It looks great. 

- You are right that the statistics for AIS survey are not available in the public data. 
- Thanks for pointing this out. Please apply country-specific commands for this survey to use s302a-y for skilled providers.

Regards, 
Aline

**Comment by yining-sun** [comment]
@wengxyu1030 Thanks Aline, 

I have replaced m2a - m2m by s302a-e, s302x-y using country-specific conditions, and added another commit to the previous pull request. 

However, as the birth.dta is not able to be generated in the first place, the related variables in the final dataset still has 0 obs. Thus, the micro-data is not updated. Following our previous discussion on similar data source problems, I did not change the master do-file's working flow.

Let me know if you have any questions or thoughts. 

Best, 
Yining
